runtime/doc/lsp.txt

```diff
@@ -812,61 +813,90 @@ client_is_stopped({client_id})                   *vim.lsp.client_is_stopped()*
         (`boolean`) stopped true if client is stopped, false otherwise.
 
 commands                                                    *vim.lsp.commands*
-    Registry for client side commands. This is an extension point for plugins
-    to handle custom commands which are not part of the core language server
-    protocol specification.
-
-    The registry is a table where the key is a unique command name, and the
-    value is a function which is called if any LSP action (code action, code
-    lenses, ...) triggers the command.
-
-    If an LSP response contains a command for which no matching entry is
-    available in this registry, the command will be executed via the LSP
-    server using `workspace/executeCommand`.
-
-    The first argument to the function will be the `Command`: Command title:
-    String command: String arguments?: any[]
+    Registry (a table) for client-side handlers, for custom server-commands
+    that are not in the LSP specification.
+
+    If an LSP response contains a command which is not found in
+    `vim.lsp.commands`, the command will be executed via the LSP server using
+    `workspace/executeCommand`.
+
+    Each key in the table is a unique command name, and each value is a
+    function which is called when an LSP action (code action, code lenses,
+    …) triggers the command.
+    • Argument 1 is the `Command`: >
+      Command
+      title: String
+      command: String
+      arguments?: any[]
+<
+    • Argument 2 is the |lsp-handler| `ctx`.
 
-    The second argument is the `ctx` of |lsp-handler|
+    Example: >lua
+        vim.lsp.commands['java.action.generateToStringPrompt'] = function(_, ctx)
```

Comment:
@mfussenegger attempt to a vim.lsp.commands example here

if you have any comments, I will fix in a followup.